cocos/network/SocketIO.cpp

```diff
@@ -35,12 +35,319 @@
+	std::string _endpoint;//
+	std::string _endpointseperator;//socket.io 1.x requires a ',' between endpoint and payload
+	std::string _type;//message type
+	std::string _separator;//for stringify the object
+	std::vector<std::string> _types;//types of messages
+};
+
+class SocketIOPacketV10x : public SocketIOPacket
+{
+public:
+	SocketIOPacketV10x();
+	virtual ~SocketIOPacketV10x();
+	int typeAsNumber();
+	std::string stringify();
+private:
+	std::vector<std::string> _typesMessage;
+};
+
+SocketIOPacket::SocketIOPacket()
+{
+	_type = "";//message type
+	_separator = ":";//for stringify the object
+	_endpointseperator = "";//socket.io 1.x requires a ',' between endpoint and payload
+	_pId = "";//id message
+	_ack = "";//
+	_name = "";//event name
+	_endpoint = "";//
+	_types.push_back("disconnect");
+	_types.push_back("connect");
+	_types.push_back("heartbeat");
+	_types.push_back("message");
+	_types.push_back("json");
+	_types.push_back("event");
+	_types.push_back("ack");
+	_types.push_back("error");
+	_types.push_back("noop");
+}
+
+SocketIOPacket::~SocketIOPacket()
+{
+	_types.clear();
+	_type = "";
+	_pId = "";
+	_name = "";
+	_ack = "";
+	_endpoint = "";
+}
+
+void SocketIOPacket::initWithType(std::string packetType)
+{
+	_type = packetType;
+}
+void SocketIOPacket::initWithTypeIndex(int index)
+{
+	_type = _types.at(index);
+}
+
+std::string SocketIOPacket::toString()
+{
+	std::stringstream encoded;
+	encoded << this->typeAsNumber();
+	encoded << this->_separator;
+
+	std::string pIdL = _pId;
+	if (_ack == "data")
+	{
+		pIdL += "+";
+	}
+
+	// Do not write pid for acknowledgements
+	if (_type != "ack")
+	{
+		encoded << pIdL;
+	}
+	encoded << this->_separator;
+
+	// Add the endpoint for the namespace to be used if not the default namespace "" or "/", and as long as it is not an ACK, heartbeat, or disconnect packet
+	if (_endpoint != "/" && _endpoint != "" && _type != "ack" && _type != "heartbeat" && _type != "disconnect") {
+		encoded << _endpoint << _endpointseperator;
+	}
+	encoded << this->_separator;
+
+	
+	if (_args.size() != 0)
+	{
+		std::string ackpId = "";
+		// This is an acknowledgement packet, so, prepend the ack pid to the data
+		if (_type == "ack")
+		{
+			ackpId += pIdL + "+";
+		}
+
+		encoded << ackpId << this->stringify();
+	}
+	
+	return encoded.str();
+}
+int SocketIOPacket::typeAsNumber()
+{
+	int num = 0;
+	std::vector<std::string>::iterator item = std::find(_types.begin(), _types.end(), _type);
+	if (item != _types.end())
+	{
+		num = item - _types.begin();
+	}
+	return num;
+}
+std::string SocketIOPacket::typeForIndex(int index)
+{
+	return _types.at(index);
+}
+
+void SocketIOPacket::addData(std::string data)
+{
+	
+	this->_args.push_back(data);
+
+}
+
+std::string SocketIOPacket::stringify()
+{
+
+std::string outS;
+if (_type == "message") {
+	outS = _args[0];
+}
+else {
+
+	rapidjson::StringBuffer s;
+	rapidjson::Writer<rapidjson::StringBuffer> writer(s);
+
+	writer.StartObject();
+	writer.Key("name");
```

Comment:
I closed this PR and opened another on the v3 branch, it includes the
changes you mention above
On Apr 24, 2015 1:46 AM, "3ltu" <notifications@github.com> wrote:

> In cocos/network/SocketIO.cpp
> <https://github.com/cocos2d/cocos2d-x/pull/11441#discussion_r29024950>:
>
> > +}
> > +
> > +std::string SocketIOPacket::stringify()
> > +{
> > +
> > +std::string outS;
> > +if (_type == "message") {
> > +	outS = _args[0];
> > +}
> > +else {
> > +
> > +	rapidjson::StringBuffer s;
> > +	rapidjson::Writer<rapidjson::StringBuffer> writer(s);
> > +
> > +	writer.StartObject();
> > +	writer.Key("name");
>
> No member Named Key in the default 3rd party cocos2d json library
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cocos2d/cocos2d-x/pull/11441/files#r29024950>.
>